src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy

```diff
@@ -1634,11 +1701,21 @@ class ActionDelegate {
              ])
          },
          uri           : { cl ->
-             item = workflowService.findOne(item.stringId)
              def uri = cl() as String
-             item.setUriNodeId(uriService.findByUri(uri).id)
-             workflowService.save(item)
+             if (item.stringId == useCase.stringId) {
```

Comment:
I think this could be improved. If-else only changes the value of the first parameter of `moveMenuItem`. It would be better to show that by calling `moveMenuItem` always after the if-else and use local variable that will be changed by it:
```groovy
def aCase = useCase
if (item.stringId != useCase.stringId) {
   aCase = workflowService.findOne(item.stringId)
}
moveMenuItem(aCase, uri)
```

Reply:
done